### ISSUE TITLE
Single-site dense ctm_tensor never updates its corners on the default SVD projector path (random/near-product A)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#723]
> 🤖 **AI-generated issue** — written by Claude Code, posted by @yingjerkao.

Found as a side observation while closing #718 (PR #722). Not a blocker for that work — the #718 fix does not depend on `ctm_tensor` — but it looks like a real defect on a default code path, so filing it rather than losing it.

## Observation

For a random asymmetric single-site `A`, `ctm_tensor(A, chi=...)` returns corners that are **bit-identical to `initialize_ctm_tensor_env`'s corners** — normalised difference exactly `0.00e+00`, all four corners, for every `max_iter` from 1 to 100, every χ from 4 to 12, at D=2 and D=3, and for more than one random `A`. The four corners also stay exactly equal to each other and exactly symmetric, while the **edges do evolve**.

It is specific to the default projector:

| `projector_method` | ‖C − C_init‖ (normalised, per corner) | ‖C − Cᵀ‖/‖C‖ |
|---|---|---|
| `"svd"` (default) | `0, 0, 0, 0` | `0, 0, 0, 0` |
| `"eigh"` | `1.35, 1.41, 0.60, 0.94` | `0.91, 1.28, 0.86, 1.02` |

Consequences visible downstream: `max_truncation_error` comes back `0.0` on the dense SVD path where the docstring says it should be meaningful, and the energy is χ-independent to twelve digits (`0.274333366574` at both χ=4 and χ=8) — the signature of an environment that never grew.

## Reproducer

```python
import jax, jax.numpy as jnp
jax.config.update("jax_enable_x64", True)
import sys; sys.path.insert(0, "tests")
from tenax.algorithms._ctm_tensor_convergence import ctm_tensor
from tenax.algorithms._ctm_tensor_init import initialize_ctm_tensor_env
from test_ctm_root_implicit_asym import _site_tensor

A = _site_tensor()          # random D=2, d=2, seed 42
for method in ("svd", "eigh"):
    init = initialize_ctm_tensor_env(A, 4)
    env, eps = ctm_tensor(A, chi=4, max_iter=100, conv_tol=1e-12,
                          projector_method=method)
    drift = []
    for n in ("C1", "C2", "C3", "C4"):
        a = jnp.asarray(getattr(init, n).todense())
        b = jnp.asarray(getattr(env, n).todense())
        drift.append(float(jnp.linalg.norm(a / jnp.linalg.norm(a)
                                           - b / jnp.linalg.norm(b))))
    print(method, ["%.2e" % x for x in drift], "eps_T", float(eps))
```

## Caveat on scope

This is on **random / near-product** input, which is already known to be a bad regime for the SVD projector — the half-infinite matrix is rank-deficient there. So the underlying cause may be a legitimate rank-1 collapse rather than a wiring bug, and physical (SU-initialised) states may be unaffected. What makes it worth a look anyway is that it fails **silently**: no warning, `eps_T = 0.0`, a plausible-looking χ-independent energy, and corners frozen at initialisation on the *default* path.

Worth checking whether the physical path is affected, and if not, whether this regime should warn rather than return a frozen environment. Related: #666 (corner collapse on near-degenerate SU states), #700/#702 (silent single-site regressions).

**Comment by yingjerkao** [comment]
> 🤖 **AI-generated comment** — written by Claude Code, posted by @yingjerkao.

## Answered: the physical SU path is affected identically. The scope caveat is wrong.

The filed caveat — *"this is on random / near-product input … the underlying cause may be a legitimate rank-1 collapse rather than a wiring bug, and physical (SU-initialised) states may be unaffected"* — does not hold. The freeze does not depend on `A` at all. It is structural in the 1x1 projector recipe, and it fires on the very first move.

### Root cause

The Fishman cross-product is

```python
# src/tenax/algorithms/_ctm_projector.py:981
M = C1g_dense.conj().T @ C4g_dense
```

`M` is indexed by the two **outer χ legs** (`col1 × col2` = χ × χ); the χ·D² `fused` index is summed away. The corners are then rebuilt as `C1' = diag(√S)·U†`, `C4' = diag(√S)·V†`.

`initialize_ctm_tensor_env` uses variPEPS `chi_init=1` semantics — `C = e₀e₀ᵀ`, `T[0, diag, 0] = 1` (`_ctm_tensor_init.py:188` and `:221`; the `SymmetricTensor` corner at `:426` does the same). Both outer legs are therefore rank-1, so `M` is rank-1, so `U = V = e₀` up to a null completion whose singular values are zero, so

**`C1' ∝ e₀e₀ᵀ` — the initial corner again.** The rank-1 corner is an *exact fixed point* of the Fishman corner update, and it regenerates itself every sweep. The environment is pinned at χ_eff = 1 forever.

Measured on a single left move, physical SU state, χ=8:

```
C1g shape (32, 8), nonzero columns: 1
M = C1g^H C4g:  shape (8, 8), rank 1
S = [2.000e+00, 0, 0, 0, 0, 0]
P_1 shape (32, 8), nonzero columns: 1,   eps_T = 0.000e+00
C1 after one move: rank 1,  ‖Ĉ1' − Ĉ1‖ = 0.000e+00
T4 after one move: 1 nonzero chi-slot of 64
```

Every symptom in the original report follows from this one fact: corners frozen bit-exactly, corners mutually equal and exactly symmetric (all of them are `e₀e₀ᵀ`), `eps_T = 0` (one nonzero singular value, so top-χ truncates nothing), and a χ-independent energy.

### The physical SU state

D=2 sublattice-rotated Heisenberg, 60 simple-update steps, default `projector_method="svd"`:

| χ | corner drift | corner rank | E |
|---|---|---|---|
| 4 | `0.00e+00` ×4 | 1 | −0.4857451736 |
| 8 | `0.00e+00` ×4 | 1 | −0.4857451736 |
| 16 | `0.00e+00` ×4 | 1 | −0.4857451736 |

Bit-identical to initialisation, energy identical to 12 digits. Indistinguishable from the random-`A` case.

### Rank-1 is not physics

Same state, same χ, 1x1 vs the 2x2 recipe — C1 singular values, normalised:

| recipe | χ=8 C1 spectrum | rank |
|---|---|---|
| 2x2 | `1, 0.128, 0.127, 0.0164, 2.08e-3, 2.02e-3` | 6 |
| 1x1 | `1, 0, 0, 0, 0, 0` | 1 |

The state has a perfectly ordinary corner spectrum with the expected degenerate pairs. The 1x1 path throws it away.

**Control arm:** perturbing the initial env off rank-1 (adding 1e-3 noise to every corner and edge) lets the 1x1 corners escape to rank 3 on the same `A`. So the trigger is the initialiser, not the input.

### Blast radius

| path | status |
|---|---|
| single-site dense, `svd` (**default**) | **frozen** |
| single-site dense, `eigh` | escapes |
| single-site dense, `qr` | escapes |
| single-site `SymmetricTensor`, `svd` (**default**) — called by `fpeps()` at `fermionic_ipeps.py:460` | **frozen** |
| multisite `recipe="1x1"` | **frozen** |
| multisite `recipe="2x2"` (**default**, incl. `iPEPSConfig.gs_recipe`) | escapes |
| root-implicit `_ctm_root_implicit_asym` (#715 / #718 / #721) | escapes — rank 4 at χ=4, 7–8 at χ=8, converges in 23–25 iters |

The 2x2 recipe escapes because its `M` retains a (χ·D²) index on **both** sides, so D² directions are available at step 1 and the environment bootstraps out of χ_init=1 as variPEPS intends. That is why this never showed up in the benchmarks: `gs_recipe` defaults to `"2x2"`. `gs_recipe="1x1"` is a supported option that silently yields a χ_eff=1 environment.

### Consequence

2x2 recipe as reference, same SU state at each D:

| | 2x2 | 1x1 `svd` (frozen) | error |
|---|---|---|---|
| D=2, χ = 4 / 8 / 16 | −0.4886385046 | −0.4857451736 | 2.9e-3 |
| D=3, χ = 9 / 16 / 24 | −0.4914466274 | **−0.1876587519** | **0.304 (62%)** |

Identical to all printed digits across χ in every row. At D=2 the frozen environment looks nearly harmless — that is an accident of a weakly-entangled state where a χ_eff=1 boundary is a fair approximation. At D=3 it is simply wrong, and no χ rescues it. (Both absolute energies sit above the true 2D Heisenberg value ≈ −0.6694 because these are modest simple-update states, not optimised ones; the quantity of interest is the 1x1-vs-2x2 gap.)

### `eigh` / `qr` are not a workaround

They escape the freeze but do not converge to anything trustworthy on the same state: E = −0.506 / −0.736 / −0.263 at χ = 4 / 8 / 16, non-monotonic, with corner ranks diverging per corner (16/2/8/10) at χ=16. They escape via arbitrary null-space directions of a rank-deficient decomposition — accident, not design. Consistent with #666.

### Two related findings

1. **`eps_T ≡ 0` is structural on this path, not a reporting gap.** `M` is χ×χ, so it has at most χ singular values and top-χ truncation discards nothing — by construction, for any input. The docstring at `_ctm_tensor_convergence.py:642` claims `max_truncation_error` is meaningful on the dense SVD path; on the 1x1 recipe it can never be nonzero, which also means the variPEPS §2.8.2 auto-χ trigger can never fire there.

2. **The split-CTM path shows the same rank-1 corners** on a physical SU state (ranks `[1,1,1,1]` at χ=4 and χ=8 after 40 sweeps). `tests/test_regularized_svd.py:247` currently documents this as expected — *"the boundary corner of a uniform 1-site iPEPS is genuinely rank-1"* — and a gradient test is written around that assumption. The 2x2 spectrum above contradicts it on the same state. Probably the same bug rationalised as physics; belongs with #463 rather than here.

### Suggested fix direction

Either seed χ_eff = D² in the initialiser so `M` has rank to work with, or route the 1x1 recipe through a projector whose cross-product retains the fused index (as 2x2 does). A regression test should assert corner **rank > 1** after convergence on a physical state — corner drift alone would pass under `eigh`, which escapes for the wrong reason.

### Scope of verification

Run directly: the random-`A` reproducer, the D=2 and D=3 SU states, the one-move instrumentation, the perturbed-init control, all six rows of the blast-radius table, the split-CTM check.

Inferred rather than run end-to-end: `fpeps()` is affected because it calls `ctm_tensor` with the default `svd` and the symmetric single-site `svd` path freezes on a symmetric `A` — a full fermionic physical state was not run. Likewise the `gs_recipe="1x1"` claim rests on `ctm_tensor_2site(recipe="1x1")` freezing, which exercises the same move code, rather than on running `optimize_gs_ad` itself.
